homeassistant/components/homematicip_cloud/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow to configure HomematicIP Cloud."""
+import json
+
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+
+from .const import DOMAIN, _LOGGER, CONF_ACCESSPOINT, CONF_PIN, CONF_NAME
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class HomematicipCloudFlowHandler(data_entry_flow.FlowHandler):
+    """Config flow HomematicIP Cloud."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize HomematicIP Cloud config flow."""
+        self.hmip_auth = None
+        self.hmip_apid = None
+        self.hmip_name = None
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        from homematicip.home import Home
+        from homematicip.auth import Auth
+
+        errors = {}
+
+        if user_input is not None:
+            if user_input[CONF_ACCESSPOINT]:
+                self.hmip_apid = user_input[CONF_ACCESSPOINT]
+                self.hmip_name = user_input[CONF_NAME]
+                _LOGGER.info("Create new authtoken for %s", self.hmip_apid)
+
+                # Create new authtoken for the accesspoint
+                # Threaded version needs to be migrated to async
+                home = Home()
+                home.init(self.hmip_apid)
+                self.hmip_auth = Auth(home)
+                if user_input[CONF_PIN]:
+                    self.hmip_auth.pin = user_input[CONF_PIN]
+                res = self.hmip_auth.connectionRequest(self.hmip_apid,
+                                                       'HomeAssistant')
+
+                # Connection established
+                if res.status_code == 200:
+                    _LOGGER.info("Connection established")
+                    return await self.async_step_link()
+
+                error_code = json.loads(res.text)['errorCode']
+                if error_code == 'INVALID_PIN':
+                    errors['base'] = 'invalid_pin'
```

Comment:
If the error concerns a specific form key, use the same key in `errors` as for the form.
```py
errors[CONF_PIN] = 'invalid_pin'
```
This will make the error message show next to the concerned form slot.